cmake/Modules/FindARPACK.cmake

```diff
@@ -0,0 +1,152 @@
+#CMake module to find arpack-ng libraries
+#
+#
+#
+#
+#       ARPACK_INCLUDE_DIR
+#       ARPACK_LIBRARIES
+#       PARPACK_INCLUDE_DIR
+#       PARPACK_LIBRARIES
+#
+#
+#
+#
+# If the user defined arpack library locations
+IF(ARPACK_LIBRARIES AND ARPACK_INCLUDE_DIR)
+  SET(ARPACK_FOUND TRUE)
+  MESSAGE(STATUS "Provided arpack include path ${ARPACK_INCLUDE_DIR}")
+  MESSAGE(STATUS "Provided arpack libraries ${ARPACK_LIBRARIES}")
+ELSE()
+  SET(ARPACK_FOUND FALSE)
+ENDIF()
+
+# If MPI is found, the user also has to supply parpack libraries
+IF(MPI_FOUND AND PARPACK_LIBRARIES AND PARPACK_INCLUDE_DIR)
+  SET(PARPACK_FOUND TRUE)
+  MESSAGE(STATUS "Provided parpack include path ${PARPACK_INCLUDE_DIR}")
+  MESSAGE(STATUS "Provided parpack libraries ${PARPACK_LIBRARIES}")
+  # Both libraries are defined, return
+  IF(ARPACK_FOUND AND PARPACK_FOUND)
+    RETURN()
+  ENDIF()
+ELSE()
+  #Parpack is not needed and arpack is provided, return
+  IF(NOT MPI_FOUND AND ARPACK_FOUND)
+    RETURN()
+  ELSE()
+  # ARPACK_LIBRARIES or ARPACK_INCLUDE_DIR are not defined
```

Comment:
Potential typo? Should this read the following?
``` suggestion
  # PARPACK_LIBRARIES or PARPACK_INCLUDE_DIR are not defined
```

Reply:
You are right, it's a typo